test/helper/setup.bash

```diff
@@ -28,51 +31,58 @@ function __initialize_variables() {
+# Print the private config path for the given container or test file,
+# if no container name was given.
+#
+# @param ${1} = container name [OPTIONAL]
+function _print_private_config_path() {
+  local TARGET_NAME=${1:-$(basename "${BATS_TEST_FILENAME}")}
+  echo "${REPOSITORY_ROOT}/test/duplicate_configs/${TARGET_NAME}"
 }
 
-# TODO: Should also fail early on "docker logs ${1} | egrep '^[  FATAL  ]'"?
-# @param ${1} name of the postfix container
-function wait_for_finished_setup_in_container() {
-  local STATUS=0
-  repeat_until_success_or_timeout --fatal-test "container_is_running ${1}" "${TEST_TIMEOUT_IN_SECONDS}" sh -c "docker logs ${1} | grep 'is up and running'" || STATUS=1
 
-  if [[ ${STATUS} -eq 1 ]]; then
-    echo "Last ${NUMBER_OF_LOG_LINES} lines of container \`${1}\`'s log"
-    docker logs "${1}" | tail -n "${NUMBER_OF_LOG_LINES}"
+# Create a dedicated configuration directory for a test file.
+#
+# @param ${1} = relative source in test/config folder
+# @param ${2} = (optional) container name, defaults to ${BATS_TEST_FILENAME}
+# @return     = path to the folder where the config is duplicated
+function _duplicate_config_for_container() {
+  local OUTPUT_FOLDER
+  OUTPUT_FOLDER=$(_print_private_config_path "${2}")
+
+  if [[ -z ${OUTPUT_FOLDER} ]]
+  then
+    echo "'OUTPUT_FOLDER' in '_duplicate_config_for_container' is empty" >&2
+    return 1
   fi
 
-  return ${STATUS}
+  rm -rf "${OUTPUT_FOLDER:?}/"
+  mkdir -p "${OUTPUT_FOLDER}"
+  cp -r "${REPOSITORY_ROOT}/test/config/${1:?}/." "${OUTPUT_FOLDER}" || return $?
+
+  echo "${OUTPUT_FOLDER}"
 }
 
 # Common defaults appropriate for most tests, override vars in each test when necessary.
 # For all tests override in `setup_file()` via an `export` var.
 # For individual test override the var via `local` var instead.
 #
 # For example, if you need an immutable config volume that can't be affected by other tests
-# in the file, then use `local TEST_TMP_CONFIG=$(duplicate_config_for_container . "${UNIQUE_ID_HERE}")`
-#
-# REQUIRED: `CONTAINER_NAME` must be set before this method is called.
-# It only affects the `TEST_TMP_CONFIG` directory created,
-# but will be used in `common_container_create()` and implicitly in other helper methods.
-function init_with_defaults() {
+# in the file, then use `local TEST_TMP_CONFIG=$(_duplicate_config_for_container . "${UNIQUE_ID_HERE}")`
+function _init_with_defaults() {
   __initialize_variables
 
   export TEST_TMP_CONFIG
-  TEST_TMP_CONFIG=$(duplicate_config_for_container . "${CONTAINER_NAME}")
+
+  # In `setup_file()` the default name to use for the currently tested docker container
+  # is `${CONTAINER_NAME}` global defined here. It derives the name from the test filename:
+  # `basename` to ignore absolute dir path and file extension, only extract filename.
+  # In `setup_file()` creates a single copy of the test config folder to use for an entire test file:
+  TEST_TMP_CONFIG=$(_duplicate_config_for_container . "${CONTAINER_NAME}")
```

Comment:
This looks like you applied this before the relevant PR was updated and merged. Comments appear to be outdated here.

Reply:
Yep, sorry!

Reply:
Done